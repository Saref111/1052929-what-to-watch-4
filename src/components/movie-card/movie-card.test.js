import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

const details = {
  bgPoster: `BG_HREF`,
  cover: `POSTER_HREF`,
  genre: `comedy`,
  year: 2001,
  rate: 8.8,
  votes: 666,
  director: `Kevin Smith`,
  actors: [`actor1`, `actor2`],
  description: {
    prescription: `prescription`,
    postscription: `postscription`,
  },
};

describe(`Testing MovieCard component`, () => {
  test(`Snapshot test`, ()=> {
    const onCardHoverHandler = jest.fn();
    const onHeaderClickHandler = jest.fn();
    const tree = renderer.create(
        <MovieCard
          movie={{
            id: 1,
            title: `title`,
            src: `src`,
            preview: `preview`,
            details,
          }}
          onCardHoverHandler={onCardHoverHandler}
          onHeaderClickHandler={onHeaderClickHandler}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
