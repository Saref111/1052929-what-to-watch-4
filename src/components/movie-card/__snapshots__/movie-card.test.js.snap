// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieCard component Snapshot test 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOut={[Function]}
  onMouseOver={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      className="player__video"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      title
    </a>
  </h3>
</article>
`;
